lib/glific/flows/action.ex

```diff
@@ -488,6 +508,50 @@ defmodule Glific.Flows.Action do
     ContactAction.send_broadcast(context, action, messages)
   end
 
+  def execute(%{type: "start_session"} = action, context, _messages) do
+    flow_uuid = action.flow["uuid"]
+
+    # check if we've seen this flow in this execution
+    if Map.has_key?(context.uuids_seen, flow_uuid) do
+      Glific.log_error("Repeated loop, hence finished the flow", false)
+    else
+      # if the action is part of a terminal node, then lets mark this context as
+      # complete, and use the parent context
+      {:node, node} = context.uuid_map[action.node_uuid]
+
+      {context, parent_id} =
+        if node.is_terminal == true,
+          do:
+            {FlowContext.reset_one_context(context,
+               source: "start_session",
+               event_meta: %{
+                 "action" => "#{inspect(action)}",
+                 "current_flow_uuid" => context.flow_uuid,
+                 "new_flow" => flow_uuid
+               }
+             ), context.parent_id},
+          else: {context, context.id}
```

Comment:
I think we already have the code somewhere, why we are just copy pasting that code ?